ow_power_system/include/PredictionHandler.h

```diff
@@ -0,0 +1,38 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef __PREDICTION_HANDLER_H__
+#define __PREDICTION_HANDLER_H__
+
+#include <vector>
+#include <map>
+
+#include "Messages/MessageBus.h"
+#include "Messages/ProgEventMessage.h"
+#include "Messages/ScalarMessage.h"
+#include "ModelBasedAsyncPrognoserBuilder.h"
+
+// The PredictionHandler class subscribes to the battery EoD event message and
+// prints each event as it is received.
+class PredictionHandler : public IMessageProcessor
+{
+public:
```

Comment:
Good to be explicit about the copy constructor and assignment operator, .e.g. define them or use `= delete` or `= default`.